### DISCUSSION TITLE
Why there are so many rules depending one spaces, paranthesis and etc...

### DISCUSSION THREAD

**Discussion by safarsafarov** [airbnb/swift#185] [General]

**Comment by fdiaz** [comment]
Edit: This was posted before the title was changed from _How is Airbnb.com using iOS universal links from the same domain?_  to _Why there are so many rules depending one spaces, paranthesis and etc..._

---

Hi @safarsafarov !

This repo is mostly for our Swift Style Guide, but I'd highly recommend you to ask questions like this on the [Mobile Native Foundation](https://github.com/MobileNativeFoundation/discussions/discussions) repository (which Airbnb is part of).

There are some topics already in terms of deep linking, like https://github.com/MobileNativeFoundation/discussions/discussions/144 where @bachand [talked about how we handle some of this in our app](https://github.com/MobileNativeFoundation/discussions/discussions/144#discussioncomment-2008969).

**Comment by calda** [comment]
We require that all style rules that affect the format of code (e.g. spacing, parenthesis, indentation, general syntax -- anything that can be changed without affecting the behavior of the code itself) can be autocorrected using SwiftFormat or SwiftLint. Rules related to spacing, parenthesis, etc are all pretty trivial to autocorrect, so we tend to adopt rules for these pieces of syntax. This somewhat biases our style guide to be largely focused around syntax (spaces, parens, indentation, file organization, redundant code) as opposed to other sorts of loose recommendations around code patterns or larger-scale app best practices.
